components/brave_wallet_ui/common/slices/entities/blockchain-token.entity.ts

```diff
@@ -237,15 +237,13 @@ export const makeSelectUserAssetIdsForChainIdFromQueryResult = () => {
   )
 }
 
-export const makeSelectAllUserAssetsForChainFromQueryResult = () => {
-  return createDraftSafeSelector(
-    [
-      selectTokensRegistryFromQueryResult,
-      (_, chainId: EntityId) => chainId
-    ],
-    (registry, chainId) => getEntitiesListFromEntityState(registry, registry.idsByChainId[chainId])
-  )
-}
+export const selectAllUserAssetsByChainIdFromQueryResult = createDraftSafeSelector(
```

Comment:
This should remain as a function that creates a selector. Otherwise the selector cache will invalidate every time a new chainID is passed into the selector. components using selectors that involve props should create a selector instance per component instance.

See: 
- https://redux-toolkit.js.org/rtk-query/api/created-api/endpoints#example-1
- https://redux-toolkit.js.org/api/createSelector